src/components/Image/Image.tsx

```diff
@@ -1,49 +1,73 @@
+import React, {
+  PropsWithChildren,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from "react";
 import ContentLoader from "components/ContentLoader";
 import ExtLink from "components/ExtLink";
 import ImagePlaceholder from "./ImagePlaceholder";
 
-export type ImageProps = {
-  src?: string;
+export type ImageProps = React.ImgHTMLAttributes<HTMLImageElement> & {
+  fallbackSrc?: string;
   isSrcLoading?: boolean;
 } & ({ href: string; title: string } | { href?: never; title?: never });
 
-type Props = {
-  img?: ImageProps;
-  className?: string;
-};
+const Image = React.forwardRef<HTMLImageElement, ImageProps>(
+  ({ className, alt = "", ...props }, forwardRef) => {
+    const ref = useRef<HTMLImageElement>(null);
+    const [isLoading, setLoading] = useState(!!props.src || props.isSrcLoading);
+    const [isError, setError] = useState(false);
 
-export default function Image({ img, className }: Props) {
-  const ref = useRef<HTMLImageElement>(null);
-  const [isLoading, setLoading] = useState(!!img?.src || img?.isSrcLoading);
+    useImperativeHandle<HTMLImageElement | null, HTMLImageElement | null>(
+      forwardRef,
+      () => ref.current
+    );
 
-  if (!img?.src) {
-    return <ImagePlaceholder className={className} />;
-  }
+    if (!props.src || isError) {
+      return <ImagePlaceholder className={className} />;
+    }
 
-  const shouldLoad = !ref.current?.complete && isLoading;
+    const shouldLoad = !ref.current?.complete && isLoading;
 
-  return (
-    <>
-      {shouldLoad && <ContentLoader className={className} />}
-      {!img.isSrcLoading && (
-        <WithLink
-          className={`${className} ${shouldLoad ? "hidden" : ""}`}
-          href={img.href}
-          title={img.title}
-        >
-          <img
-            ref={ref}
-            src={img.src}
-            className={`${className} object-contain w-full h-full`}
-            alt=""
-            onLoad={() => setLoading(false)}
-          />
-        </WithLink>
-      )}
-    </>
-  );
-}
+    return (
+      <>
+        {shouldLoad && <ContentLoader className={className} />}
+        {!props.isSrcLoading && (
+          /**
+           * Setting the logic to add `hidden` class name is necessary on both
+           * `WithLink` wrapper and on the child `img`.
+           *
+           * Reason:
+           * if no `href` was passed, that means only the image would be returned and since
+           * it is returned without a wrapper, we need to apply `hidden` className manually.
+           * Otherwise (if `href` was passed), we need to apply `hidden` to the link component
+           * wrapping the `img`.
+           */
+          <WithLink
+            className={`${className} ${shouldLoad ? "hidden" : ""}`}
+            href={props.href}
+            title={props.title}
+          >
+            <img
+              ref={ref}
+              src={props.src}
+              className={`object-contain ${
```

Comment:
Most of the components that call `Image` use `object-contain` as one of their classes, so we can put it as one of the default classes. All components that need to use some other [object-fit](https://tailwindcss.com/docs/object-fit) style need just pass it and it will overwrite this class name.